### PR TITLE
Remove no longer necessary selection checks

### PR DESCRIPTION
**Pull request by mchowning**
Remove checks that are now handled on the native side by Android.

See the related gutenberg-mobile PR for further description and testing steps: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2279

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/rich-text/src/component/index.native.js [hunk 1]
```diff
@@ -722,51 +722,31 @@ export class RichText extends Component {
...
+							newSelectionStart = 0;
+						}
+						let newSelectionEnd =
+							this.props.selectionEnd - count;
+						if ( newSelectionEnd < 0 ) {
+							newSelectionEnd = 0;
 						}
+						selection = {
+							start: newSelectionStart,
+							end: newSelectionEnd,
```
Comment (mchowning) [inline packages/rich-text/src/component/index.native.js:748]: Of the three selection checks, this is the only one that triest to do something more intelligent than just setting the selection to `null` if there is a problem, so I decided to leave it in even though I think we could probably remove it as well now.

**Comment by geriux** [comment]
We could remove [this](https://github.com/WordPress/gutenberg/blob/40fb567d7623de69a7eb08930d6fc322fad6ee95/packages/rich-text/src/component/index.native.js#L634) and [this](https://github.com/WordPress/gutenberg/blob/40fb567d7623de69a7eb08930d6fc322fad6ee95/packages/rich-text/src/component/test/index.native.js#L13) as well right? What do you think?

**Comment by mchowning** [comment]
> We could remove this and this as well right? What do you think?

You're absolutely right! I forgot to remove those, but I have now. Thanks!

**Review by geriux — APPROVED** [review]
LGTM! It's always good to see some code refactor =)

**Pull request merged**